src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLeaseStore.cs

```diff
@@ -73,9 +78,10 @@ public async Task<bool> ReleaseInitializationLockAsync()
         {
             string lockId = this.GetStoreLockName();
             Uri documentUri = UriFactory.CreateDocumentUri(this.leaseStoreCollectionInfo.DatabaseName, this.leaseStoreCollectionInfo.CollectionName, lockId);
+            var documentServiceLease = DocumentServiceLease.FromDocument(new Document { Id = lockId });
+            this.requestOptionsFactory.AddPartitionKeyIfNeeded(documentServiceLease, lockId);
```

Comment:
If the PK value, when added, should match the Id, why do we need to pass it as a second parameter? Cannot we extract it from the first parameter?

Reply:
Michael didnt want pk to match id value in all cases. e.g when we create a new lease, he wanted it to be guid.NewGuid.
to keep method interface same between 2 overloads, i am asking for pk value even in this method.